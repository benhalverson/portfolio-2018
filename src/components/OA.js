import React, { Fragment } from 'react';

import OAdefaultview from '../images/OA-default-view.png';
import OAresults from '../images/OA-results.png';
import OAchart from '../images/OA-chart.png';
import OAtablechart from '../images/OA-table-chart.png';

const OA = () => (
  <Fragment>
    <div className="title">E-Trade Options Analyzer</div>
    <div className="body">
      <p className="description">
        I worked on this project with one other engineer. We rebuilt it from the
        ground up using Angular.js, a forked version of bootstrap and
        ui-bootstrap. I primarly worked on the charting part of the UI.
      </p>
      <div className="screenshots">
        <img className="col" src={OAdefaultview} alt="" />
        <img className="col" src={OAresults} alt="" />
        <img className="col" src={OAchart} alt="" />
        <img className="col" src={OAtablechart} alt="" />
      </div>
    </div>
  </Fragment>
);

export default OA;
