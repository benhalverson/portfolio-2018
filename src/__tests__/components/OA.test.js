import React from 'react';
import { shallow } from 'enzyme';
import OA from '../../components/OA';
test('it should render the GSKVaccines component', () => {
  const wrapper = shallow(<OA />);
  expect(wrapper).toMatchSnapshot();
});
