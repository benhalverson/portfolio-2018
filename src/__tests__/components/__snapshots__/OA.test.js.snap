// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the GSKVaccines component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OA />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
>
          E-Trade Options Analyzer
</div>,
        <div
          className="body"
>
          <p
                    className="description"
          >
                    I worked on this project with one other engineer. We rebuilt it from the ground up using Angular.js, a forked version of bootstrap and ui-bootstrap. I primarly worked on the charting part of the UI.
          </p>
          <div
                    className="screenshots"
          >
                    <img
                              alt=""
                              className="col"
                              src="OA-default-view.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="OA-results.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="OA-chart.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="OA-table-chart.png"
                    />
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "E-Trade Options Analyzer",
          "className": "title",
        },
        "ref": null,
        "rendered": "E-Trade Options Analyzer",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description"
>
              I worked on this project with one other engineer. We rebuilt it from the ground up using Angular.js, a forked version of bootstrap and ui-bootstrap. I primarly worked on the charting part of the UI.
</p>,
            <div
              className="screenshots"
>
              <img
                            alt=""
                            className="col"
                            src="OA-default-view.png"
              />
              <img
                            alt=""
                            className="col"
                            src="OA-results.png"
              />
              <img
                            alt=""
                            className="col"
                            src="OA-chart.png"
              />
              <img
                            alt=""
                            className="col"
                            src="OA-table-chart.png"
              />
</div>,
          ],
          "className": "body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "I worked on this project with one other engineer. We rebuilt it from the ground up using Angular.js, a forked version of bootstrap and ui-bootstrap. I primarly worked on the charting part of the UI.",
              "className": "description",
            },
            "ref": null,
            "rendered": "I worked on this project with one other engineer. We rebuilt it from the ground up using Angular.js, a forked version of bootstrap and ui-bootstrap. I primarly worked on the charting part of the UI.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="col"
                  src="OA-default-view.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="OA-results.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="OA-chart.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="OA-table-chart.png"
/>,
              ],
              "className": "screenshots",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OA-default-view.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OA-results.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OA-chart.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OA-table-chart.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
>
            E-Trade Options Analyzer
</div>,
          <div
            className="body"
>
            <p
                        className="description"
            >
                        I worked on this project with one other engineer. We rebuilt it from the ground up using Angular.js, a forked version of bootstrap and ui-bootstrap. I primarly worked on the charting part of the UI.
            </p>
            <div
                        className="screenshots"
            >
                        <img
                                    alt=""
                                    className="col"
                                    src="OA-default-view.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="OA-results.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="OA-chart.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="OA-table-chart.png"
                        />
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "E-Trade Options Analyzer",
            "className": "title",
          },
          "ref": null,
          "rendered": "E-Trade Options Analyzer",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description"
>
                I worked on this project with one other engineer. We rebuilt it from the ground up using Angular.js, a forked version of bootstrap and ui-bootstrap. I primarly worked on the charting part of the UI.
</p>,
              <div
                className="screenshots"
>
                <img
                                alt=""
                                className="col"
                                src="OA-default-view.png"
                />
                <img
                                alt=""
                                className="col"
                                src="OA-results.png"
                />
                <img
                                alt=""
                                className="col"
                                src="OA-chart.png"
                />
                <img
                                alt=""
                                className="col"
                                src="OA-table-chart.png"
                />
</div>,
            ],
            "className": "body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "I worked on this project with one other engineer. We rebuilt it from the ground up using Angular.js, a forked version of bootstrap and ui-bootstrap. I primarly worked on the charting part of the UI.",
                "className": "description",
              },
              "ref": null,
              "rendered": "I worked on this project with one other engineer. We rebuilt it from the ground up using Angular.js, a forked version of bootstrap and ui-bootstrap. I primarly worked on the charting part of the UI.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    className="col"
                    src="OA-default-view.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="OA-results.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="OA-chart.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="OA-table-chart.png"
/>,
                ],
                "className": "screenshots",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OA-default-view.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OA-results.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OA-chart.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OA-table-chart.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
